op-e2e/actions/proofs/l1_osaka_fork_test.go

```diff
@@ -0,0 +1,158 @@
+
+		atBlockWithHash := func(hash common.Hash) *bind.CallOpts {
+			return &bind.CallOpts{
+				BlockHash: hash,
+			}
+		}
+
+		// requireConsistentBlobBaseFeeForFork requires the blob base fee to be consistent between
+		// the L1 Origin block (computed using the excess blob gas and l1 chain config)
+		// and the L1 Block contract on L2 (accessed with a contract method call), for a given fork predicate.
+		requireConsistentBlobBaseFeeForFork := func(t actionsHelpers.StatefulTesting, l2Block eth.L2BlockRef, expectActive bool, label string, isActive func(num *big.Int, time uint64) bool) {
+			bbfL2, err := l1BlockContract.BlobBaseFee(atBlockWithHash(l2Block.Hash))
+			require.NoError(t, err)
+
+			l1Origin := miner.L1Chain().GetHeaderByHash(l2Block.L1Origin.Hash)
+			if expectActive {
+				require.True(t, isActive(l1Origin.Number, l1Origin.Time), "%s not active at l1 origin %d, time %d", label, l1Origin.Number, l1Origin.Time)
+			} else {
+				require.False(t, isActive(l1Origin.Number, l1Origin.Time), "%s should not be active at l1 origin %d, time %d", label, l1Origin.Number, l1Origin.Time)
+			}
+			bbfL1 := eip4844.CalcBlobFee(env.Sd.L1Cfg.Config, l1Origin)
+
+			require.True(t, bbfL2.Cmp(bbfL1) == 0,
+				"%s: blob base fee does not match, bbfL2=%d, bbfL1=%d, l1BlockNum=%d, l2BlockNum=%d", label, bbfL2, bbfL1, l1Origin.Number, l2Block.Number)
+
+			require.True(t, bbfL2.Cmp(big.NewInt(1)) > 0,
+				"%s: blob base fee is unrealistically low and doesn't exercise the blob fee calculation", label)
+		}
+
+		// buildL1ToTime advances L1 with empty blocks until the given fork time.
+		buildL1ToTime := func(t actionsHelpers.StatefulTesting, forkTime *uint64) {
+			require.NotNil(t, forkTime, "fork time must be configured")
+			h := miner.L1Chain().CurrentHeader()
+			for h.Time < *forkTime {
+				h = miner.ActEmptyBlock(t).Header()
+			}
+		}
+
+		// Build L1 blocks to trigger Fusaka activation
+		// The blob parameters don't change between Prague and Osaka.
+		// So this test is only useful when we hit the BPO forks below
+		l1Block := miner.ActBuildToOsaka(t)
+		require.Equal(t, uint64(2), l1Block.Number().Uint64())
+
+		// Build an empty L2 block which has a pre-Fusaka L1 origin, and check the blob fee is correct
+		sequencer.ActL2EmptyBlock(t)
+		l2Block := sequencer.SyncStatus().UnsafeL2
+		require.Equal(t, uint64(1), l2Block.Number)
+		requireConsistentBlobBaseFeeForFork(t, l2Block, false, "Osaka", func(num *big.Int, time uint64) bool { return env.Sd.L1Cfg.Config.IsOsaka(num, time) })
+
+		// Advance L2 chain until L1 origin has Fusaka activ
+		sequencer.ActL1HeadSignal(t)
+		sequencer.ActBuildToL1HeadUnsafe(t)
+
+		// Check that the L1 origin is now a Fusaka block, and that the blob fee is correct
+		l2Block = sequencer.L2Unsafe()
+		require.Greater(t, l2Block.Number, uint64(1))
+		requireConsistentBlobBaseFeeForFork(t, l2Block, true, "Osaka", func(num *big.Int, time uint64) bool { return env.Sd.L1Cfg.Config.IsOsaka(num, time) })
```

Comment:
can't this be deduped into the loop below?

Reply:
Yes, did that here [cc37aff](https://github.com/ethereum-optimism/optimism/pull/17666/commits/cc37affc2167c7333688cae611656941cfeecfd3)